ground/src/main/java/com/google/android/ground/MainViewModel.kt

```diff
@@ -115,6 +116,15 @@ constructor(
       onUserSignInError(e)
     }
 
+  private suspend fun getTos(): TermsOfService? =
+    try {
+      termsOfServiceRepository.getTermsOfService()
+    } catch (e: Error) {
+      // Request timed out or failed due to some error (e.g. PERMISSION_DENIED)
+      Timber.e(e)
+      null
```

Comment:
Wouldn't this mean users can skip accepting the ToS if they fail to load for any reason?

Reply:
This was the behavior so far. Do you think we should just error out if the request fails?

Reply:
Fixed